packages/nimble-components/src/table-column/mixins/fractional-width-column.ts

```diff
@@ -0,0 +1,55 @@
+import { attr, nullableNumberConverter } from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+type SizedTableColumn = Pick<
+TableColumn,
+'internalFractionalWidth' | 'internalMinPixelWidth' | 'defaultMinPixelWidth'
+>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type SizedTableColumnConstructor = abstract new (...args: any[]) => SizedTableColumn;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function fractionalWidthColumn<
+    TBase extends SizedTableColumnConstructor
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to support being resized
+     * proportionally within a Table.
+     */
+    abstract class FractionalWidthColumn extends base {
+        public fractionalWidth?: number | null = 1;
+
+        public minPixelWidth?: number | null;
```

Comment:
Seems like both of these default, i.e. `1` and `undefined` should be exported and shared from table-column/base/types.ts as well